### PR TITLE
Add TciOnFdGrid to ValenciaDivClean

### PR DESCRIPTION
**Pull request by nilsdeppe**
## Proposed changes

The TCI that checks if the DG solution would be okay again

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `major new feature` if appropriate.

### CODE REVIEW

File: src/Evolution/Systems/GrMhd/ValenciaDivClean/Subcell/TciOnFdGrid.hpp [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  using argument_tags =
+      tmpl::list<evolution::dg::subcell::Tags::Inactive<
+                     grmhd::ValenciaDivClean::Tags::TildeD>,
+                 evolution::dg::subcell::Tags::Inactive<
+                     grmhd::ValenciaDivClean::Tags::TildeTau>,
+                 grmhd::ValenciaDivClean::Tags::VariablesNeededFixing,
+                 domain::Tags::Mesh<3>>;
+  static bool apply(const Scalar<DataVector>& tilde_d,
+                    const Scalar<DataVector>& tilde_tau,
+                    bool vars_needed_fixing, const Mesh<3> dg_mesh,
```
Comment (kidder) [inline src/Evolution/Systems/GrMhd/ValenciaDivClean/Subcell/TciOnFdGrid.hpp]: take Mesh by reference?

File: src/Evolution/Systems/GrMhd/ValenciaDivClean/Subcell/TciOnFdGrid.hpp [hunk 2]
```diff
@@ -0,0 +1,58 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "DataStructures/VariablesTag.hpp"
```
Comment (kidder) [inline src/Evolution/Systems/GrMhd/ValenciaDivClean/Subcell/TciOnFdGrid.hpp]: unused

File: src/Evolution/Systems/GrMhd/ValenciaDivClean/Subcell/TciOnFdGrid.hpp [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+#include "Domain/Tags.hpp"
+#include "Evolution/DgSubcell/Tags/Inactive.hpp"
+#include "Evolution/DgSubcell/Tags/OnSubcells.hpp"
```
Comment (kidder) [inline src/Evolution/Systems/GrMhd/ValenciaDivClean/Subcell/TciOnFdGrid.hpp]: unused

File: src/Evolution/Systems/GrMhd/ValenciaDivClean/Subcell/TciOnFdGrid.hpp [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+#include "Evolution/Systems/GrMhd/ValenciaDivClean/FixConservatives.hpp"
```
Comment (kidder) [inline src/Evolution/Systems/GrMhd/ValenciaDivClean/Subcell/TciOnFdGrid.hpp]: unused

File: src/Evolution/Systems/GrMhd/ValenciaDivClean/Subcell/TciOnFdGrid.hpp [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+#include "Evolution/Systems/GrMhd/ValenciaDivClean/Tags.hpp"
+#include "Evolution/VariableFixing/Tags.hpp"
+#include "PointwiseFunctions/GeneralRelativity/TagsDeclarations.hpp"
```
Comment (kidder) [inline src/Evolution/Systems/GrMhd/ValenciaDivClean/Subcell/TciOnFdGrid.hpp]: unused

File: src/Evolution/Systems/GrMhd/ValenciaDivClean/Subcell/TciOnFdGrid.hpp [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+#include "PointwiseFunctions/Hydro/EquationsOfState/EquationOfState.hpp"
```
Comment (kidder) [inline src/Evolution/Systems/GrMhd/ValenciaDivClean/Subcell/TciOnFdGrid.hpp]: unused

File: src/Evolution/Systems/GrMhd/ValenciaDivClean/Subcell/TciOnFdGrid.hpp [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+#include "PointwiseFunctions/Hydro/Tags.hpp"
```
Comment (kidder) [inline src/Evolution/Systems/GrMhd/ValenciaDivClean/Subcell/TciOnFdGrid.hpp]: unused

File: src/Evolution/Systems/GrMhd/ValenciaDivClean/Subcell/TciOnFdGrid.hpp [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+#include "Utilities/TMPL.hpp"
+
+/// \cond
+class DataVector;
+template <size_t Dim>
+class Mesh;
+namespace gsl {
```
Comment (kidder) [inline src/Evolution/Systems/GrMhd/ValenciaDivClean/Subcell/TciOnFdGrid.hpp]: unused

File: src/Evolution/Systems/GrMhd/ValenciaDivClean/Subcell/TciOnFdGrid.cpp [hunk 9]
```diff
@@ -0,0 +1,24 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Evolution/Systems/GrMhd/ValenciaDivClean/Subcell/TciOnFdGrid.hpp"
+
+#include <cstddef>
```
Comment (kidder) [inline src/Evolution/Systems/GrMhd/ValenciaDivClean/Subcell/TciOnFdGrid.cpp:6]: unused

File: tests/Unit/Evolution/Systems/GrMhd/ValenciaDivClean/Subcell/Test_TciOnFdGrid.cpp [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+#include <cstddef>
+#include <memory>
+
+#include "DataStructures/DataBox/DataBox.hpp"
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Domain/Tags.hpp"
+#include "Evolution/DgSubcell/Mesh.hpp"
+#include "Evolution/Systems/GrMhd/ValenciaDivClean/Subcell/TciOnFdGrid.hpp"
+#include "Evolution/Systems/GrMhd/ValenciaDivClean/System.hpp"
```
Comment (kidder) [inline tests/Unit/Evolution/Systems/GrMhd/ValenciaDivClean/Subcell/Test_TciOnFdGrid.cpp]: System.hpp -> Tags.hpp

File: tests/Unit/Evolution/Systems/GrMhd/ValenciaDivClean/Subcell/Test_TciOnFdGrid.cpp [hunk 11]
```diff
@@ -0,0 +1,72 @@
...
+#include "Framework/TestingFramework.hpp"
+
+#include <cstddef>
+#include <memory>
+
+#include "DataStructures/DataBox/DataBox.hpp"
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Domain/Tags.hpp"
+#include "Evolution/DgSubcell/Mesh.hpp"
```
Comment (kidder) [inline tests/Unit/Evolution/Systems/GrMhd/ValenciaDivClean/Subcell/Test_TciOnFdGrid.cpp]: unused

File: tests/Unit/Evolution/Systems/GrMhd/ValenciaDivClean/Subcell/Test_TciOnFdGrid.cpp [hunk 12]
```diff
@@ -0,0 +1,72 @@
...
+  const Mesh<3> mesh{6, Spectral::Basis::Legendre,
+                     Spectral::Quadrature::GaussLobatto};
+  const double persson_exponent = 5.0;
+
+  auto box = db::create<
+      db::AddSimpleTags<evolution::dg::subcell::Tags::Inactive<
+                            grmhd::ValenciaDivClean::Tags::TildeD>,
+                        evolution::dg::subcell::Tags::Inactive<
+                            grmhd::ValenciaDivClean::Tags::TildeTau>,
+                        grmhd::ValenciaDivClean::Tags::VariablesNeededFixing,
```
Comment (kidder) [inline tests/Unit/Evolution/Systems/GrMhd/ValenciaDivClean/Subcell/Test_TciOnFdGrid.cpp:32]: include for this tag

**Review by kidder — CHANGES_REQUESTED** [review]

File: src/Evolution/Systems/GrMhd/ValenciaDivClean/Subcell/TciOnFdGrid.cpp (hunk 9 above)
Comment (nilsdeppe) [reply src/Evolution/Systems/GrMhd/ValenciaDivClean/Subcell/TciOnFdGrid.cpp:6]: Mesh has a size_t as a template parameter :shrug:

**Review by nilsdeppe — COMMENTED** [review]
I've rebased on develop and pushed a fixup. Thanks for the review :)

**Comment by kidder** [comment]
you can squash

**Comment by nilsdeppe** [comment]
okay, squashed

**Review by kidder — APPROVED** [review]

**Review by fmahebert — APPROVED** [review]

**Pull request merged**